cmd/kube-proxy/app/server.go

```diff
@@ -620,6 +620,10 @@ func (s *ProxyServer) Run() error {
 	endpointsConfig.RegisterEventHandler(s.Proxier)
 	go endpointsConfig.Run(wait.NeverStop)
 
+	nodeConfig := config.NewNodeConfig(informerFactory.Core().V1().Nodes(), s.ConfigSyncPeriod)
```

Comment:
We were talking about watching only ObjectMetadata of nodes, right?
Watching whole Node objects (which can be really big) is really expensive.

Reply:
This alpha specifically does not assume scalability.  EndpointSlice will mean we no longer need to watch nodes at all (except "this node" which can be a selector), right?

Reply:
Yes - watching "this node" is fine.

If this is explicitly excluding scalability, I'm fine with a TODO (but I think it deserves at least a TODO here).

Reply:
I just tested watching PartialObjectMetadata which we discussed in https://github.com/kubernetes/enhancements/pull/1086#issuecomment-514494392 It indeed will also be updated frequently (resource version). 

Currently we only cache and compare the node labels in kube-proxy when watch node, If we switch to watch PartialObjectMetadata of node, I think will it not significantly improve performance, but will be better than watch node, I'm trying to switch to watch PartialObjectMetadata of node. 

Here is my test code I mentioned above:
``` go
package main

import (
	"flag"
	"fmt"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/apimachinery/pkg/runtime/schema"
	"k8s.io/client-go/metadata"
	"k8s.io/client-go/metadata/metadatainformer"
	"k8s.io/client-go/tools/cache"
	"k8s.io/client-go/tools/clientcmd"
	"k8s.io/klog"
	"os"
	"reflect"
	"time"
)

func main() {
	config, err := clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
	if err != nil {
		panic(err.Error())
	}

	metadataClient, err := metadata.NewForConfig(config)
	if err != nil {
		panic(err.Error())
	}

	fs := flag.NewFlagSet("glog", flag.ExitOnError)
	klog.InitFlags(fs)
	fs.Parse(os.Args[1:])

	gvr := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "nodes"}
	metadataInformer := metadatainformer.NewFilteredMetadataInformer(metadataClient, gvr, metav1.NamespaceAll, 15*time.Minute, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, nil)
	metadataInformer.Informer().AddEventHandlerWithResyncPeriod(
		cache.ResourceEventHandlerFuncs{
			AddFunc: func(obj interface{}) {
				key, err := cache.MetaNamespaceKeyFunc(obj)
				if err == nil {
					fmt.Println("add ", key)
				}
			},
			UpdateFunc: func(old interface{}, new interface{}) {
				key, err := cache.MetaNamespaceKeyFunc(new)
				if err == nil {
					fmt.Println("update ", key)
				}
				fmt.Println("deepequal:", reflect.DeepEqual(old, new))
			},
			DeleteFunc: func(obj interface{}) {
				key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
				if err == nil {
					fmt.Println("delete ", key)
				}
			},
		},
		15*time.Minute,
	)
	stop := make(chan struct{})
	metadataInformer.Informer().Run(stop)
}
```

Reply:
So there still is not an api to watch node labels without getting triggered frequently? Is there an open issue for that I could follow?

Reply:
Not found yet.

Reply:
Some related things:
https://github.com/kubernetes/kubernetes/issues/29772
https://github.com/kubernetes/kubernetes/issues/14733
https://github.com/kubernetes/kubernetes/issues/80289